src/Core/Content/Product/Stock/OrderStockSubscriber.php

```diff
@@ -48,7 +52,12 @@ public static function getSubscribedEvents(): array
 
     public function beforeWriteOrderItems(EntityWriteEvent $event): void
     {
-        if (!$this->enableStockManagement) {
+        $enableStockManagement = $this->enableStockManagement;
+        if (Feature::isActive('v6.7.0.0')) {
+            $enableStockManagement = $this->systemConfigService->getBool('core.listing.enableStockManagement') ?? true;
```

Comment:
$enableStockManagement = (bool) $this->systemConfigService->get('core.listing.enableStockManagement') ?? true;